### DISCUSSION TITLE
Call Java function from C#

### DISCUSSION THREAD

**Discussion by nguyenhothanhtam0709** [ikvmnet/ikvm#324] [General]
I have a question that: Can I write some function in Java code, build to .jar file and call there functions in C#? If it is possible. Can you give me an example? Thanks.

**Comment by wasabii** [comment]
That is pretty much the entire point of IKVM, yes. I think the README amply covers this at this time.

**Comment by nguyenhothanhtam0709** [comment]
@wasabii , I used IkvmReference to include the .jar file. But when executed  **dotnet run**, I got this error.
![image](https://github.com/ikvmnet/ikvm/assets/90300565/fed5824a-d40e-4879-add1-8d8cf22ac4ac)
I'm running on Mac m2.

**Comment by wasabii** [comment]
We don't currently support Apple ARM64 processors. Might be able to run the thing in X64, if you can do that.

**Reply by nguyenhothanhtam0709** [reply]
@wasabii , Thanks for this information. I'll try on x64.

**Reply by wasabii** [reply]
This is due to our build process being stuck on .NET Core 3.1 currently. .NET Core 3.1 has no OS X ARM64 support.

**Reply by nguyenhothanhtam0709** [reply]
@wasabii, Does Ikvm support .NET core 6.0 on Ubuntu x64?

**Reply by wasabii** [reply]
Yes.

The problem is we can't build against anything above .NET 3.1. The results will RUN though.

**Reply by nguyenhothanhtam0709** [reply]
@wasabii , Thanks for answering me.
